lib/trie.js

```diff
@@ -0,0 +1,66 @@
+module.exports = Trie;
+
+function Trie() {
+  this.methods = {};
+}
+
+//insert new nodes to Trie
+Trie.prototype.add = function(path, verb, handler) {
+  var node = this;
+  var parts = getPathParts(path);
+  if (parts.length) {
+    for (var i = 0; i < parts.length; i++) {
+      if (!(parts[i] in node)) {
```

Comment:
please extract `parts[i]` into a variable to avoid repetition, e.g. `var segment = parts[i]`. Or rename `parts` to `pathParts` and then use `var part = pathParts[i]`.